azure/durable_functions/decorators/durable_app.py

```diff
@@ -0,0 +1,228 @@
+
+            return decorator()
+
+        return wrap
+
+    def activity_trigger(self, input_name: str,
+                         activity: Optional[str] = None):
+        """Register an Activity Function.
+
+        Parameters
+        ----------
+        input_name: str
+            Parameter name of the Activity input.
+        activity: Optional[str]
+            Name of Activity Function.
+            The value is None by default, in which case the name of the method is used.
+        """
+        @self._configure_function_builder
+        def wrap(fb):
+            def decorator():
+                fb.add_trigger(
+                    trigger=ActivityTrigger(name=input_name,
+                                            activity=activity))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+    def entity_trigger(self, context_name: str,
+                       entity_name: Optional[str] = None):
+        """Register an Entity Function.
+
+        Parameters
+        ----------
+        context_name: str
+            Parameter name of the Entity input.
+        entity_name: Optional[str]
+            Name of Entity Function.
+            The value is None by default, in which case the name of the method is used.
+        """
+        @self._configure_entity_callable
+        @self._configure_function_builder
+        def wrap(fb):
+            def decorator():
+                fb.add_trigger(
+                    trigger=EntityTrigger(name=context_name,
+                                          entity_name=entity_name))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+    def _add_rich_client(self, fb, parameter_name,
+                         client_constructor):
+        # Obtain user-code and force type annotation on the client-binding parameter to be `str`.
+        # This ensures a passing type-check of that specific parameter,
+        # circumventing a limitation of the worker in type-checking rich DF Client objects.
+        # TODO: Once rich-binding type checking is possible, remove the annotation change.
+        user_code = fb._function._func
+        user_code.__annotations__[parameter_name] = str
+
+        # `wraps` This ensures we re-export the same method-signature as the decorated method
+        @wraps(user_code)
+        async def df_client_middleware(*args, **kwargs):
+
+            # Obtain JSON-string currently passed as DF Client,
+            # construct rich object from it,
+            # and assign parameter to that rich object
+            starter = kwargs[parameter_name]
+            client = client_constructor(starter)
+            kwargs[parameter_name] = client
+
+            # Invoke user code with rich DF Client binding
+            return await user_code(*args, **kwargs)
+
+        user_code_with_rich_client = df_client_middleware
+        fb._function._func = user_code_with_rich_client
+
+    def durable_client_input(self,
```

Comment:
any tests for these decorators?

Reply:
since non binding type check is not yet supported in worker, can we add comment here to inform that if rich client type injection is to be released?

Reply:
@YunchuWang.

I just added a test for the DF client. It's rather minimal, but I believe it suffices for our scenario: I'm simply testing that a rich DF Client object is received.

Reply:
Can you please clarify what is your request regarding a comment? I do have an internal "TODO" comment about enabling type-checking here: https://github.com/Azure/azure-functions-durable-python/blob/c888b48590d4e4ecbed92be6bc07ead9ed0f9332/azure/durable_functions/decorators/durable_app.py#L166-L173

Reply:
Is that comment above sufficient for you^?